### PR TITLE
Various minor fixes

### PR DESCRIPTION
**Pull request by WillB97**
To form release 2025.1.1

### CODE REVIEW

File: scripts/run_simulator.py [hunk 1]
```diff
@@ -18,11 +19,22 @@
...
     # Assume the script is in the scripts directory
-    SIM_BASE = Path(__file__).parents[1]
+    SIM_BASE = Path(__file__).parents[1].resolve()
+
+POSSIBLE_WEBOTS_PATHS = [
+    ("darwin", "/Applications/Webots.app/Contents/MacOS/webots"),
+    ("win32", "C:\\Program Files\\Webots\\msys64\\mingw64\\bin\\webotsw.exe"),
+    ("win32", expandvars("%LOCALAPPDATA%\\Programs\\Webots\\msys64\\mingw64\\bin\\webotsw.exe")),
+    # Attempt to use the start menu shortcut
+    ("win32", expandvars("%ProgramData%\\Microsoft\\Windows\\Start Menu\\Programs\\Cyberbotics\\Webots.lnk")),
```
Comment (antonnikitin97) [inline scripts/run_simulator.py:33]: Nothing wrong here  - just commenting to say this is interesting, I didn't know this was how you handled environment vars in Python... neat.

File: scripts/run_simulator.py [hunk 2]
```diff
@@ -36,46 +48,26 @@ def get_webots_parameters() -> tuple[Path, Path]:
...
-            possible_paths = [
-                "C:\\Program Files\\Webots\\msys64\\mingw64\\bin\\webotsw.exe",
-                expandvars("%LOCALAPPDATA%\\Programs\\Webots\\msys64\\mingw64\\bin\\webotsw.exe"),
-            ]
-            for path in possible_paths:
+        for system_filter, path in POSSIBLE_WEBOTS_PATHS:
+            if sys.platform.startswith(system_filter):
+                print(f"Checking {path}")
                 if Path(path).exists():
                     webots = path
```
Comment (antonnikitin97) [inline scripts/run_simulator.py:64]: Silly question, why can't we do `return Path(webots), world_file` here?

Comment (WillB97) [reply scripts/run_simulator.py:64]: We'd still need an else clause to handle returning when L56 does find an executable, but essentially either would be fine.

Comment (antonnikitin97) [reply scripts/run_simulator.py:64]: Ah ok, I see :)

File: scripts/run_simulator.py [hunk 3]
```diff
@@ -36,46 +48,26 @@ def get_webots_parameters() -> tuple[Path, Path]:
     if not world_file.exists():
         raise RuntimeError("World file not found.")
 
+    if not (SIM_BASE / "venv").exists():
+        print("Please run the setup.py script before running the simulator.")
+        raise RuntimeError
```
Comment (RealOrangeOne) [inline scripts/run_simulator.py:53]: Keeping this to one line will mean the message will be at the bottom of the terminal.
```suggestion
        raise RuntimeError("Please run the setup.py script before running the simulator.")
```

**Review by RealOrangeOne — APPROVED** [review]

**Pull request merged**